### PR TITLE
CONSOLE-3372, OCPBUGS-13648, OCPBUGS-13833: Console supports installing non-latest Operator versions

### PR DESCRIPTION
**Pull request by sg00dwin**
https://issues.redhat.com/browse/CONSOLE-3372

Enable the user to select from available channels and associated versions when installing an Operator. 

Note that "installing an older version" means "no automatic update"; hence, when users select a non-latest Operator version, this implies the "Update" field would be changed to "Manual".

Remaining items
- [x] Update installation version in the modal header when choosing a non latest channel.
- [x] Tests

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx:104]: Needs i18n.

**Review by rhamilto — COMMENTED** [review]
I gave this a cursory look and added comments, but would definitely prefer @jhadvig do the final review since I've been involved in the development.

File: frontend/packages/operator-lifecycle-manager/src/types.ts
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/types.ts:260]: I think this should fix the TS errors you're seeing in CI
```suggestion
      entries?: {
```

**Review by jhadvig — APPROVED** [review]
Really nice work @sg00dwin 👍 
/lgtm
/approve

**Comment by sg00dwin** [comment]
QE Approver:
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

**Comment by RickJWagner** [comment]
Looks good.

/label px-approved

**Comment by sg00dwin** [comment]
Awaiting QE approval @yapei

**Comment by yapei** [comment]
@XiyunZhao  is testing the PR and will give feedback

**Comment by XiyunZhao** [comment]
Hi @sg00dwin, the testing is finished now, [bug](https://issues.redhat.com/browse/OCPBUGS-13142) and comment has been updated to [JIRA](https://issues.redhat.com/browse/CONSOLE-3372), please help to take a look, contact me if you have any question.

**Comment by sg00dwin** [comment]
@jhadvig  I have come across an issue when a non latest version* is installed. 
The `PackageManifest` returns `installState: Not Installed` and `installed: false`. 
*Note each channel has a default latest version known as its `currentCSV`
cc @spadgett 


This causes a couple of UX issues.

_Issue 1_ 
On the Install Operator details modal, when an operator is installed, it should display an `Alert` and the Install button should be changed to Uninstall
```
This Operator has been installed on the cluster. View it here.
```

In the case of an installed non latest version, the user is allowed to continue to the next screen.
Which is the subscribe form. It's there that an alert warning says ...
```
A Subscription for this Operator already exists in Namespace "openshift-operators"
Remove the existing Subscription in order to install this Operator in Namespace "openshift-operators"
```
and the Install button is disabled.

_Issue 2_
The "Installed" badge isn't displayed on the card.



**Bug example**
Data Grid 
8.4.x Channel has 5 versions associated 8.4.4, 8.4.3, 8.4.2, 8.4.1, 8.4.0
8.4.4 is the latest version
The installed version below is 8.4.3

operator-not-latest-version-shows-uninstalled

**Comment by XiyunZhao** [comment]
@sg00dwin Here is the page that should reflect CSV data per selected channel/version
image

**Comment by XiyunZhao** [comment]
@sg00dwin @jhadvig is there any conclusion about the "Installed" label issue?
Ref: https://github.com/openshift/console/pull/12743#issuecomment-1536796391

**Comment by XiyunZhao** [comment]
The changes look good to me, all critical acceptance is achieved
/label qe-approved

**Comment by sg00dwin** [comment]
@jhadvig This pr is ready for another look. 
**Updates**

1. Installed value uses `subscription?.status?.installedCSV` instead of `subscription?.status?.currentCSV` so that the Installed badge, filter by installed operators and alert message with Uninstall button are now displayed are noted correctly on the details screen.
2. On subscribe form, changing the version to any non-latest version switches "Update Approval" from Automatic to Manual.
3. Add Tony's text addition in the alert to better inform the user manual approval applies to all operators in a namespace.

**Review by jhadvig — APPROVED** [review]

**Review by jhadvig — APPROVED** [review]
/lgtm

**Pull request merged**
